src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -0,0 +1,679 @@
+                        || "α_2 = α[254]",
+                        self.canon_advices[2],
+                        offset,
+                        || alpha_2.ok_or(Error::SynthesisError),
+                    )?;
+                }
+
+                // Offset 2
+                {
+                    let offset = 2;
+                    // Copy z_13_alpha_0_prime
+                    copy(
+                        &mut region,
+                        || "Copy z_13_alpha_0_prime",
+                        self.canon_advices[0],
+                        offset,
+                        &z_13_alpha_0_prime,
+                        perm,
+                    )?;
+
+                    // Copy z_44_alpha
+                    copy(
+                        &mut region,
+                        || "Copy z_44_alpha",
+                        self.canon_advices[1],
+                        offset,
+                        &z_44_alpha,
+                        perm,
+                    )?;
+
+                    // Copy z_43_alpha
+                    copy(
+                        &mut region,
+                        || "Copy z_43_alpha",
+                        self.canon_advices[2],
+                        offset,
+                        &z_43_alpha,
+                        perm,
+                    )?;
+                }
+
+                Ok(())
+            },
+        )?;
+
+        Ok(result)
+    }
+
+    fn decompose_base_field_elem(
+        &self,
+        base_field_elem: CellValue<pallas::Base>,
+        offset: usize,
+        region: &mut Region<'_, pallas::Base>,
+    ) -> Result<EccBaseFieldElemFixed, Error> {
+        // Decompose base field element into 3-bit words.
+        let words: Vec<Option<u8>> = {
+            let words = base_field_elem.value().map(|base_field_elem| {
+                decompose_scalar_fixed::<pallas::Base>(
+                    base_field_elem,
+                    constants::L_ORCHARD_BASE,
+                    constants::FIXED_BASE_WINDOW_SIZE,
+                )
+            });
+
+            if let Some(words) = words {
+                words.into_iter().map(Some).collect()
+            } else {
+                vec![None; constants::NUM_WINDOWS]
+            }
+        };
+
+        // Initialize empty ArrayVec to store running sum values [z_1, ..., z_85].
+        let mut running_sum: ArrayVec<CellValue<pallas::Base>, { constants::NUM_WINDOWS }> =
+            ArrayVec::new();
+
+        // Assign running sum `z_i`, i = 0..=n, where z_{i+1} = (z_i - a_i) / (2^3)
+        // and `z_0` is initialized as `base_field_elem`.
+        let mut z = copy(
+            region,
+            || "z_0 = base_field_elem",
+            self.super_config.window,
+            offset,
+            &base_field_elem,
+            &self.super_config.perm,
+        )?;
+
+        for idx in 0..words.len() {
+            self.base_field_fixed_mul.enable(region, offset + idx)?;
+        }
+
```

Comment:
This causes the selector to be double-enabled, as it is also activated by the call to `self.super_config.assign_region_inner` above:

![image](https://user-images.githubusercontent.com/4993799/125008165-daa6ae80-e059-11eb-8b05-135d7e3f3365.png)
```suggestion
```